state/indexer/block/kv/util.go

```diff
@@ -49,13 +49,12 @@ func heightKey(height int64) ([]byte, error) {
 	)
 }
 
-func eventKey(compositeKey, typ, eventValue string, height int64, eventSeq int64) ([]byte, error) {
+func eventKey(compositeKey, eventValue string, height int64, eventSeq int64) ([]byte, error) {
 	return orderedcode.Append(
 		nil,
 		compositeKey,
 		eventValue,
 		height,
-		typ,
```

Comment:
Since this changes the format in which data is stored in the index, this is a breaking change, is it not? If so, what does the upgrade path look like for users with existing indexes from v0.34/v0.37? Do they need to reindex? If so I'd recommend adding some instructions in the `UPGRADING.md` in the backport branch that Mergify creates.

(We shouldn't have a changelog entry on `main` or extra `UPGRADING.md` instructions if this is going to be released from a backport branch)

Reply:
Thane, sorry, I am just seeing this. People will not need to re-index if they do not want because this is handled by all the if-else statements in parsing the event. But info on this has already been added to the [upgrading.md ](https://github.com/cometbft/cometbft/pull/591). Let me know if you feel we should revisit the text there.